java/pom.xml

```diff
@@ -9,6 +9,13 @@
     <build>
         <plugins>
             <plugin>
+                <artifactId>maven-compiler-plugin</artifactId>
```

Comment:
All those versions are "EOL" by Oracle. If we are going to update the version might as well update to 8.